sonar-dotnet-shared-library/src/test/java/org/sonar/plugins/dotnet/tests/ScannerFileServiceTest.java

```diff
@@ -0,0 +1,232 @@
+    when(filePredicates.all()).thenReturn(allMock);
+    when(filePredicates.and(allMock, languageKeyMock)).thenReturn(andMock);
+    when(fs.predicates()).thenReturn(filePredicates);
+
+    // act
+    ScannerFileService sut = new ScannerFileService("key", fs);
+    sut.getFilesByRelativePath("foo");
+
+    // assert
+    assertThat(argumentCaptor.getValue()).isEqualTo(andMock);
+  }
+
+  @Test
+  public void getFilesByRelativePath_when_no_indexed_files_returns_empty() {
+    FileSystem fs = createFileSystemForInputFiles(Collections.emptyList());
+
+    // act
+    ScannerFileService sut = new ScannerFileService("key", fs);
+    Optional<InputFile> result = sut.getFilesByRelativePath("C:\\_\\some\\path\\file.cs");
+
+    // assert
+    assertThat(result).isEmpty();
+    assertThat(logTester.logs(LoggerLevel.TRACE)).containsExactly(
+      "It seems Deterministic Source Paths are used, replacing 'C:\\_\\some\\path\\file.cs' with 'some\\path\\file.cs'.",
+      "Did not find any indexed file for 'some/path/file.cs'.");
+  }
+
+  @Test
+  public void getFilesByRelativePath_when_indexed_files_do_not_match_returns_empty() {
+    FileSystem fs = createFileSystemForInputFiles(Arrays.asList(mockInput("another/path/file.cs"), mockInput("some/file.cs")));
+
+    ScannerFileService sut = new ScannerFileService("key", fs);
+    Optional<InputFile> result = sut.getFilesByRelativePath("/_/some/path/file.cs");
+
+    assertThat(result).isEmpty();
+    assertThat(logTester.logs(LoggerLevel.TRACE)).containsExactly(
+      "It seems Deterministic Source Paths are used, replacing '/_/some/path/file.cs' with 'some/path/file.cs'.",
+      "Did not find any indexed file for 'some/path/file.cs'.");
+  }
+
+  @Test
+  public void getFilesByRelativePath_when_multiple_indexed_files_match_returns_empty() {
+    FileSystem fs = createFileSystemForInputFiles(Arrays.asList(mockInput("root1/some/path/file.cs"), mockInput("root2/some/path/file.cs")));
+
+    ScannerFileService sut = new ScannerFileService("key", fs);
+    Optional<InputFile> result = sut.getFilesByRelativePath("/_/some/path/file.cs");
+
+    assertThat(result).isEmpty();
+    assertThat(logTester.logs(LoggerLevel.TRACE)).containsExactly(
+      "It seems Deterministic Source Paths are used, replacing '/_/some/path/file.cs' with 'some/path/file.cs'.");
+    assertThat(logTester.logs(LoggerLevel.DEBUG)).containsExactly("Found 2 indexed files for relative path 'some/path/file.cs'. Will skip this coverage entry.");
+  }
+
+  @Test
+  public void getFilesByRelativePath_when_single_indexed_files_match_returns_file() {
+    InputFile expectedResult = mockInput("root/some/path/file.cs");
+    FileSystem fs = createFileSystemForInputFiles(Arrays.asList(
+      mockInput("one"),
+      mockInput("two"),
+      expectedResult,
+      mockInput("four")));
+
+    ScannerFileService sut = new ScannerFileService("key", fs);
+    Optional<InputFile> result = sut.getFilesByRelativePath("/_/some/path/file.cs");
```

Comment:
To demonstrate that it's not just the `root` single-level prefix, but any relative path.
```suggestion
    Optional<InputFile> result = sut.getFilesByRelativePath("/_/path/file.cs");
```